qa/rpc-tests/test_framework/mininode.py

```diff
@@ -1614,7 +1615,7 @@ class NodeConn(asyncore.dispatcher):
         "regtest": b"\xfa\xbf\xb5\xda",   # regtest
     }
 
-    def __init__(self, dstaddr, dstport, rpc, callback, net="regtest", services=NODE_NETWORK):
+    def __init__(self, dstaddr, dstport, rpc, callback, net="regtest", services=NODE_NETWORK, send_version=True):
```

Comment:
In my experience, test class and function interfaces often accumulate a lot of cruft as people add special-cases for individual test cases. It'd be nice if we could keep NodeConn's __init__() interface clean.

Would the following be a more elegant way to achieve this?

- add a send_version() method to NodeConn and move the send_version code there
- create a child class of NodeConn in your testcase that overrides the send_version() method